src/DataStructures/InfluxDBClient.cs

```diff
@@ -268,6 +270,112 @@ private async Task<bool> PostPointsAsync(string dbName, TimePrecision precision,
                 return false;
         }
 
+        private static IInfluxDatapoint ToInfluxDataPoint<T>(T point)
+        {
+            if (typeof(IInfluxDatapoint).IsAssignableFrom(typeof(T)))
+            {
+                return point as IInfluxDatapoint;
+            }
+
+            var measurementProp = typeof(T)
+               .GetProperties()
+               .Where(prop => prop.IsDefined(typeof(InfluxDBMeasurementName), true))
+               .ToList();
+            if (!measurementProp.Any())
+                throw new CustomAttributeFormatException("InfluxDBMeasurement attribute is required on object published to InfluxDB");
+            var measurementName = measurementProp.First().GetValue(point, null) as string;
+
+            var timeProp = typeof(T)
+               .GetProperties()
+               .Where(prop => prop.IsDefined(typeof(InfluxDBTime), true))
+               .ToList();
+            var time = timeProp.Any() ? (DateTime)timeProp.First().GetValue(point, null) : DateTime.Now;
```

Comment:
I think the default value should be `DateTime.UtcNow`. Influx expects the timestamp to be in UTC, and you are assigning this to `UtcTimestamp` few lines below.